Marlin/Marlin_main.cpp

```diff
@@ -6115,11 +6158,19 @@ inline void gcode_M109() {
     idle();
     refresh_cmd_timeout(); // to prevent stepper_inactive_time from running out
 
-    float temp = thermalManager.degHotend(target_extruder);
+    const float temp = thermalManager.degHotend(target_extruder);
+
+    #if ENABLED(PRINTER_EVENT_LEDS)
+      // Gradually change LED strip from violet to red as nozzle heats up
+      if (!wants_to_cool) {
+        const uint8_t blue = map(constrain(temp, start_temp, target_temp), start_temp, target_temp, 255, 0);
+        if (blue != old_blue) set_led_color(255, 0, (old_blue = blue));
+      }
+    #endif
 
```

Comment:
I'd generally reduce those two lines down to one:
```cpp
set_led_color(255, 0, blue == old_blue ? 255 : blue);
```
…but they are incorrect. First of all, `old_blue` is not being set to `blue` when it differs. Secondly, with that correction made, `blue == old_blue` will evaluate true most of the time.